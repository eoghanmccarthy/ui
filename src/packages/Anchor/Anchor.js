import React, { forwardRef } from "react";
import { string, oneOf } from "prop-types";
import cx from "classnames";

import baseProps from "../../utils/baseProps";

const Anchor = forwardRef(
  (
    { children, className, href = null, target = "_blank", ...rest },
    forwardedRef
  ) => {
    return (
      <a
        ref={forwardedRef}
        {...(href && { href: href })}
        {...(href && { target: target })}
        className={cx("ui-anchor", { placeholder: !href }, className)}
        {...rest}
      >
        {children}
      </a>
    );
  }
);

export default baseProps(Anchor);

Anchor.propTypes = {
  href: string,
  target: oneOf(["_blank", "_self", "_parent", "_top"])
};
