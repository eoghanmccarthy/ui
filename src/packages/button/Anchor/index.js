import React, { forwardRef } from "react";
import { string, oneOf } from "prop-types";
import cx from "classnames";

import ButtonBase from "../ButtonBase";

const Anchor = forwardRef(
  (
    { children, className, href = null, target = "_blank", ...rest },
    forwardedRef
  ) => {
    return (
      <a
        ref={forwardedRef}
        {...(href && { href: href })}
        {...(href && { target: target })}
        className={cx("ui-anchor", { placeholder: !href }, className)}
        {...rest}
      >
        {children}
      </a>
    );
  }
);

export default ButtonBase(Anchor);

Anchor.propTypes = {
  href: string,
  target: oneOf(["_blank", "_self", "_parent", "_top"])
};
