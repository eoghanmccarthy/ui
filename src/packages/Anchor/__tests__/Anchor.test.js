import React from "react";
import ReactDOM from "react-dom";
import { render, cleanup } from "@testing-library/react";
import userEvent from "@testing-library/user-event";
import "@testing-library/jest-dom/extend-expect";
import renderer from "react-test-renderer";

import Anchor from "../Anchor.js";

describe("Anchor component", () => {
  afterEach(cleanup);

  const MockComponent = props => (
    <Anchor testId={"instance-test-id"} {...props} />
  );

  it("renders without crashing", () => {
    const div = document.createElement("div");
    ReactDOM.render(
      <MockComponent href={"http://google.com"}>click</MockComponent>,
      div
    );
  });

  it("renders children", () => {
    const { getByTestId } = render(
      <MockComponent href={"http://google.com"}>Click</MockComponent>
    );
    expect(getByTestId("instance-test-id")).toHaveTextContent("Click");
  });

  it("renders with correct css classes", () => {
    const { getByTestId } = render(
      <MockComponent className={"instance-class"} href={"http://google.com"}>
        Click
      </MockComponent>
    );
    expect(getByTestId("instance-test-id")).toHaveClass(
      "ui-anchor",
      "instance-class"
    );
  });

  it("renders correctly as placeholder link", () => {
    const { getByTestId } = render(<MockComponent>Click</MockComponent>);
    expect(getByTestId("instance-test-id")).toHaveClass("placeholder");
  });

  // it("renders and onClick event is called", () => {
  //   const onClick = jest.fn();
  //
  //   const { getByTestId } = render(
  //     <MockComponent className={"instance-class"} onClick={onClick}>
  //       Click
  //     </MockComponent>
  //   );
  //   userEvent.click(getByTestId("instance-test-id"));
  //   expect(onClick).toHaveBeenCalled();
  // });

  it("matches snapshot", () => {
    const tree = renderer
      .create(
        <MockComponent className={"instance-class"} href={"http://google.com"}>
          Click
        </MockComponent>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
