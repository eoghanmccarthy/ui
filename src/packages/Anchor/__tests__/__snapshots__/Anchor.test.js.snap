// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Anchor component matches snapshot 1`] = `
<a
  className="ui-anchor instance-class"
  data-testid="instance-test-id"
  href="http://google.com"
  target="_blank"
>
  Click
</a>
`;
